frontend/src/components/shared/WithItemComponent.js

```diff
@@ -0,0 +1,13 @@
+import React from "react"
+import { useItem } from "../../hooks/useItem"
+
+export const WithItemComponent = (withItem) => (itemsByID, id, fn, defaultValue = null, render = (item) => <>{item}</>) => {
```

Comment:
This function could use some documentation to explain what it is for and how to use it.